File: queue/simplequeue/simplequeue.py

'''
	Time complexity :

		access    : O(n)
		search    : O(n)
		insertion : O(1)
		delete    : o(1) 
'''

class queue :

	def __init__(self) :
		self.array = []

	def _empty(self) :
		return True if len(self.array) == 0 else False

	def _size(self) :
		return len(self.array)

	def _enqueue(self , value) :

		self.array.append(value)

	def _dequeue(self) :

		return self.array.pop(0) if not self._empty() else "Empty Queue"

	def _front(self) :

		return self.array[0] if not self._empty() else "Empty Queue"

	def _display(self) :

		for i in self.array :
			print(i , end = " ")
		print("")

q = queue()
q._enqueue(3)
q._enqueue(4)
q._enqueue(5)
q._enqueue(6)
q._dequeue()
q._enqueue(7)
q._dequeue()

print(f"Size : {q._size()}")

q._display()

File: queue/simplequeue/simplequeue.php
<?php

    /*
        Time complexity :

			access    : O(n)
			search    : O(n)
			insertion : O(1)
			delete    : o(1) 
    */

    class queue
    {
        function __construct()
        {
            $this->array = array() ;
            $this->rear = -1 ;
            $this->head = -1 ;
            $this->size = 0 ;
        }

        public function _empty()
        {
            return ($this->rear == -1 && $this->head == -1) ? true: false;
        }

        public function _size()
        {
            return $this->size ;
        }

        public function _enqueue($value)
        {
            if($this->_empty())
            {
                $this->rear = 0 ;
                $this->head = 0 ;
            }
            else
                $this->rear++ ;
            $this->array[$this->rear] = $value ;
            $this->size++ ;
        }

        public function _dequeue()
        {
            if($this->_empty())
                return ;
            if($this->rear == $this->head)
            {
                $this->rear = -1 ;
                $this->head = -1 ;
            }
            else
                $this->head++ ;
            $this->size-- ;
        }

        public function _front()
        {
            if($this->_empty())
                return ;
            return $this->array[$this->head] ;
        }

        public function _traversal()
        {
            if($this->_empty())
                return ;
            $temp = array() ;
            while(! $this->_empty())
            {
                array_push($temp, $this->_front()) ;
                $this->_dequeue() ;
            }
            return $temp ;
        }
    }

    $q = new queue() ;
    
    $q->_dequeue() ;
    $q->_enqueue("s") ;
    $q->_enqueue("H") ;
    $q->_enqueue("e") ;
    $q->_dequeue() ;
    $q->_enqueue("l") ;
    $q->_enqueue("l") ;
    $q->_enqueue("o") ;

    echo "Front: " . $q->_front() . "\n" ;
    echo "Size: " . $q->_size() . "\n" ;
    echo join("", $q->_traversal()) . "\n" ;
?>
